test/src/test/java/org/corfudb/AbstractCorfuTest.java

```diff
@@ -102,6 +225,49 @@ protected void succeeded(Description description) {
             }
             System.out.print(ansi().a("[").fg(Ansi.Color.GREEN).a("PASS")
                     .reset().a("]" + testStatus).newline());
+            System.out.flush();
+        }
+
+        /** Print a list of the active threads.
+         *
+         */
+        protected void printThreads() {
+            System.out.println(ansi().fgCyan().bold().a("Active Threads At Test Failure")
+                .reset());
+            System.out.println(ansi().format("%-40s %s",
+                "Thread name", "Method"));
+            Map<Thread, StackTraceElement[]> threads = Thread.getAllStackTraces();
+            threads.entrySet().forEach(e -> {
+                // Don't print the current thread.
+                if (!e.getKey().equals(Thread.currentThread())) {
+                    if (e.getValue().length > 0) {
+                        System.out.println(ansi().format("%-40s %s:%s",
+                            e.getKey().getName(),
+                            e.getValue()[0].getClassName(),
+                            e.getValue()[0].getMethodName()));
+                    } else {
+                        System.out.println(ansi().format("%-40s (no trace available)",
+                            e.getKey().getName()));
+                    }
+                }
+            });
+        }
+
+        /** Print the latest logs, up to the number of log entries defined in LOG_ELEMENTS.
+         *
+         */
+        protected void printLogs() {
+            System.out.println(ansi().fgCyan().bold().a("Last ").a(LOG_ELEMENTS)
+                .a(" Logs Until Failure")
+                .reset());
+            Iterable<byte[]> logEvents = LOG_APPENDER.getEventsAndReset();
```

Comment:
is there any test jvm memory requirements change with inmemory appender.
I guess you have a bound in your appender so should be ok.

Reply:
Probably not a significant amount...